spec/requests/templates_spec.rb

```diff
@@ -29,6 +29,30 @@
     end
   end
 
+  context "editing a template" do
+    let!(:template) { FactoryGirl.create(:template, :name => 'foo', :description => 'bar') }
+    before { api_basic_authorize(action_identifier(:templates, :edit)) }
+    subject { put(api_template_url(nil, template), :params => params) }
+
+    describe "PUT /api/templates/:c_id" do
+      let(:params) { {:name => 'baz'} }
+
+      it 'edits the template name' do
+        expect { subject }.to change { template.reload.name }.to('baz')
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    describe "PATCH /api/templates/:c_id" do
+      let(:params) { {:description => 'baz'} }
+
+      it 'edits the template description' do
+        expect { subject }.to change { template.reload.description }.to('baz')
```

Comment:
subject still exercises the put above on line 35, we need a patch for this.

Reply:
Right, is it okay if I do with `send`? Or should I copy some code (that I don't like) :wink: 
```ruby
describe 'PATCH' do
  let(:req) { :patch }
  ...
end
describe 'PUT' do
  let(:req) { :put }
  ...
end
subject { send(req) ... }
```